stl/inc/experimental/random

```diff
@@ -0,0 +1,50 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};

```

Comment:
The TS says that the engine must be "initialized to an unpredictable state". Using a 32-bit seed creates the potential for collisions; I suspect that we should initialize the entire state of the Mersenne Twister.

```cpp
#include <algorithm>
#include <functional>
#include <random>
#include <stddef.h>
#include <stdint.h>
#include <vector>
using namespace std;

void initialize_randomness(mt19937& mt) {
    constexpr size_t n = mt19937::state_size;
    constexpr size_t w = mt19937::word_size;
    static_assert(w % 32 == 0, "w % 32 == 0");
    constexpr size_t k = w / 32;
    vector<uint32_t> vec(n * k);
    random_device rd;
    generate(vec.begin(), vec.end(), ref(rd));
    seed_seq seq(vec.cbegin(), vec.cend());
    mt.seed(seq);
}

#include <stdio.h>

int main() {
    mt19937 mt;
    initialize_randomness(mt);
    printf("%u\n", mt());
}
```

Reply:
Of course in the real one the size is constant so you want a array rather than a vector :)